apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Combobox } from '@headlessui/react';
+import { useRpcClient } from '@mysten/core';
+import {
+    getObjectFields,
+    getObjectType,
+    normalizeSuiObjectId,
+} from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+import clsx from 'clsx';
+import { useState } from 'react';
+
+import { ReactComponent as SearchIcon } from '~/assets/SVGIcons/24px/Search.svg';
+import {
+    getFieldTypeValue,
+    FieldValueType,
+    extractSerializationType,
+} from '~/components/ownedobjects/utils';
+import { FieldItem } from '~/components/ownedobjects/views/FieldItem';
+import { useGetObject } from '~/hooks/useGetObject';
+import { Banner } from '~/ui/Banner';
+import { DisclosureBox } from '~/ui/DisclosureBox';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
+import { Text } from '~/ui/Text';
+import { ListItem, VerticalList } from '~/ui/VerticalList';
+
+interface ObjectFieldsProps {
+    id: string;
+}
+
+export function ObjectFieldsCard({ id }: ObjectFieldsProps) {
+    const { data, isLoading, isError } = useGetObject(id);
+    const [query, setQuery] = useState('');
+    const [activeFieldName, setActiveFieldName] = useState('');
+    const objectType = getObjectType(data!);
+
+    // Get the packageId, moduleName, functionName from the objectType
+    const [packageId, moduleName, functionName] =
+        objectType?.split('<')[0]?.split('::') || [];
+    const rpcClient = useRpcClient();
+
+    // Get the normalized struct for the object
+    const {
+        data: normalizedStruct,
+        isLoading: loadingNormalizedStruct,
+        isError: errorNormalizedMoveStruct,
+    } = useQuery(
+        ['normalized-struct', id],
+        () =>
+            rpcClient.getNormalizedMoveStruct({
+                package: normalizeSuiObjectId(packageId),
+                module: moduleName,
+                struct: functionName,
+            }),
+        {
+            enabled: !!packageId && !!moduleName && !!functionName,
+        }
+    );
+
+    if (isLoading || loadingNormalizedStruct) {
+        return (
+            <div className="flex w-full justify-center">
+                <LoadingSpinner text="Loading data" />
+            </div>
+        );
+    }
+    if (isError || errorNormalizedMoveStruct) {
+        return (
+            <Banner variant="error" spacing="lg" fullWidth>
+                Failed to get field data for :{id}
+            </Banner>
+        );
+    }
+    const fieldsData = getObjectFields(data!);
+    const fieldsNames = Object.entries(fieldsData || {});
```

Comment:
Shouldn't this also use the `normalizedStruct`, given that's what's mapped over in the body panel?

Reply:
Yes,  I missed this,